### ISSUE TITLE
Ruby OS X packages built from Kokoro have a new dll dependency on /usr/local/opt/gmp/lib/libgmp.10.dylib 

### ISSUE THREAD

**Issue by apolcyn** [grpc/grpc#14999]
1.10.0 is the last release built from jenkins. The dlls dependencies of the 1.10.0 Ruby package can be viewed with:

```
$ gem fetch grpc --version 1.10.0 --platform universal-darwin
$ gem unpack grpc-1.10.0-universal-darwin.gem
$ otool -L grpc-1.10.0-universal-darwin/src/ruby/lib/grpc/2.5/grpc_c.bundle
grpc-1.10.0-universal-darwin/src/ruby/lib/grpc/2.5/grpc_c.bundle:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1213.0.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```

The 1.10.1 package (not yet released) is built from Kokoro, and the new dependency can be seen with:

```
$ # download the 1.10.1 mac package built from kokoro (currently unreleased)
$ gem unpack  grpc-1.10.1-universal-darwin.gem
$ otool -L grpc-1.10.1-universal-darwin/src/ruby/lib/grpc/2.5/grpc_c.bundle
grpc-1.10.1-universal-darwin/src/ruby/lib/grpc/2.5/grpc_c.bundle:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	/usr/local/opt/gmp/lib/libgmp.10.dylib (compatibility version 14.0.0, current version 14.2.0)
	/usr/lib/libobjc.A.dylib (compatibility version 1.0.0, current version 228.0.0)
```

where the `/usr/local/opt/gmp/lib/libgmp.10.dylib` dep is new.

I'm not yet sure what the exact cause is and how the dependency can be fixed (if not fixed `gmp` would be a new required dependency for the 1.11+ releases - it can be installed with `brew install gmp`). 

I'm also not yet certain which other grpc packages have the same issue.

cc @mehrdada @nicolasnoble @matt-kwong @jtattermusch

**Comment by jtattermusch** [comment]
Do we know if this is due to the ruby installation in kokoro workers or if it's due to some code changes between 1.10.x versions?

**Comment by apolcyn** [comment]
> Do we know if this is due to the ruby installation in kokoro workers or if it's due to some code changes between 1.10.x versions?

Also not sure yet, though the diff between the 10.0 and 10.1 is unsuspecting. I've scheduled a package build from the 1.10.0 release tag to check (build is pending at [build #450](https://kokoro2.corp.google.com/job/grpc/job/core/job/experimental/job/grpc_build_artifacts_multiplatform/))

**Comment by apolcyn** [comment]
It's starting to look like this is due to a [feature that was added in to ruby 2.1+](https://bugs.ruby-lang.org/issues/8796) which uses `libgmp` in ruby it's present on the system, and falls back to not using it if not available (there seems to be a configure option to force not using it). Presumably`libgmp` is present on kokoro mac workers but was not available on jenkins.

**Comment by matt-kwong** [comment]
Using the flag to force not using it is the best option IMO. An alternative is running `brew uninstall gmp` at the start of the build to remove the dependency.

**Comment by apolcyn** [comment]
I agree explicitly disabling would be best, looking into a way to do that.
